source/ada/lsp-ada_contexts.adb

```diff
@@ -112,31 +124,44 @@ package body LSP.Ada_Contexts is
          Ada.Directories.Search (Root, "", Dirs_Only, On_Dir'Access);
       end Search_GPR_File;
 
-      Root : constant Ada.Strings.UTF_Encoding.UTF_8_String :=
-        LSP.Types.To_UTF_8_String (Self.Root);
-
       Name : constant Ada.Strings.UTF_Encoding.UTF_8_String :=
         LSP.Types.To_UTF_8_String (File);
 
       Result : GNATCOLL.VFS.Virtual_File;
    begin
+      --  If Name was provided, search for the corresponding project
       if Name /= "" then
+
+         --  First, search the project file relatively to the root
          Result := GNATCOLL.VFS.Create_From_Base
            (Base_Dir  => GNATCOLL.VFS.Filesystem_String (Root),
             Base_Name => GNATCOLL.VFS.Filesystem_String (Name));
 
          if Result.Is_Regular_File then
             return Result;
          end if;
-      end if;
 
-      Search_GPR_File (Root, Result);
+         --  If not found, perform a comprehensive search everywhere below
+         --  root. TODO: I'm not sure about that logic. As an user I might
```

Comment:
I don't think comments in the code is right place to discuss server behavior. If you are not sure, open an issue instead of add a comment.

Reply:
I agree, however a comment is better than nothing in the meantime ! I added a comment because I needed to understand what this function did.

Reply:
I opened an issue and a reference to the issue in the comment. I prefer to keep the comment so we don't forget about it later :)

Reply:
So, we have the issue and we won't forget it. Part after TODO: is not required any more.

Reply:
Again, I disagree! It might not be apparent to somebody reading/modifying this code that there is something going on here that might need to be changed, and the information might be useful to her/him. Or, as @setton put it, there is very rarely too much comments. But whatever, will remove it, it's unimportant enough.